### DISCUSSION TITLE
Is there a way to pass props to a pages component in Nuxt 3?

### DISCUSSION THREAD

**Discussion by brantac** [nuxt/nuxt#16100] [Questions]
Is there an equivalent way to [Vue Router](https://router.vuejs.org/guide/essentials/passing-props.html) of passing data to a route component through props?

In my `index.vue` component page I have a list of articles,
```Vue
<template>
    <div>
        <h1>Hello world!</h1>
        <NuxtLink
            v-for="article in articles"
            :key="article.id"
            :to="{path: '/article/' + article.slug, query: {id: article.id}}"
            :id="article.id">
            <!-- :to="'/article/' + article.slug"> -->
            <h3>{{ article.title }}</h3>
        </NuxtLink>
    </div>
</template>
```
and I want to pass the `article.id` to the child component `/pages/article/[slug].vue`, that will be rendered when I click in one of the links.

**Comment by nathanchase** [comment]
You could just grab the id from the route directly - `route.params.slug`.

**Reply by brantac** [reply]
I actually don't want to have params in my URL, if that's possible. I want it to be just `https://[site].com/article/title-of-the-article`, instead of `https://[site].com/article/title-of-the-article?id=1`.

**Reply by nathanchase** [reply]
As far as I know, Nuxt 3 does not have a way to [send props to route components](https://router.vuejs.org/guide/essentials/passing-props.html). This would have to be an added feature hooking into Vue Router 4's `props` option. Perhaps the Nuxt team could consider adding it, or perhaps a module might be created to add the functionality.

**Reply by harlan-zw** [reply]
It does seem like a useful feature, though in your example shouldn't you just fetch based on the slug?

You'd have to do it anyway if someone visits the URL directly, if you're just trying to avoid a HTTP request, you can just store the fetched articles in a variable and query that before making the request. Though, iirc nuxt will fetch the payload regardless

**Reply by brantac** [reply]
Yeah, I might be overthinking. Fetching based on the slug it's probably the way to go. Thank you all!
